tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -701,6 +702,12 @@ bool DeclResultIdMapper::createStageVars(const DeclaratorDecl *decl,
     if (asInput && shaderModel.IsGS() && hasGSPrimitiveTypeQualifier(decl))
       qual = hlsl::DxilParamInputQual::InputPrimitive;
 
+    // Note that geometry shaders have output streams that are required to be
+    // marked as "inout". The DxilParamInputQual for these cases must be
+    // 'OutStream' rather than 'Out'.
+    if (!asInput && isOutputStream)
+      qual = hlsl::DxilParamInputQual::OutStream0;
```

Comment:
I see we use `OutStream0` here directly. We may need to leave a TODO here to mention multiple output stream support?